src/d3a/models/strategy/load_hours.py

```diff
@@ -182,14 +182,14 @@ def _one_sided_market_event_tick(self, market):
                 if max_energy < FLOATING_POINT_TOLERANCE:
                     return
                 if acceptable_offer.energy > max_energy:
+                    self.energy_requirement_Wh[market.time_slot] = 0
```

Comment:
Weird. But I guess order matters now more than before.

Reply:
Order matters due to reentrancy. I noticed in several occasions when trying to use non-blocking events, that calling this code twice from 2 different threads had the effect that the load accepted every available offer. The reason for that was the event_offer, since it calls this method, whenever a new offer is created during a trade, it is published before the trade is settled. This causes this code to be called again and to accept one more offer, even though the energy requirement was fulfilled by the not-already-settled-offer. This generated yet another trade, and the cycle went on. This was circumvented by updating the energy requirement before accepting the offer. 
With blocking events this issue no longer appears, still it's good to have it as a precaution.